File: frameworkgen/scanCard/Source/HyperswitchScancard.m
#import <React/RCTBridgeModule.h>

@interface RCT_EXTERN_MODULE(HyperswitchScancard, NSObject)

RCT_EXTERN_METHOD(launchScanCard: (NSString)rnMessage :(RCTResponseSenderBlock)rnCallback)

+ (BOOL)requiresMainQueueSetup
{
    return NO;
}

@end

File: frameworkgen/scanCard/Source/HyperswitchScancard.swift
import HyperswitchScancard

@objc(HyperswitchScancard)
class HyperswitchScancard: NSObject {
    
    @objc
    func launchScanCard(_ rnMessage: String, _ rnCallback: @escaping RCTResponseSenderBlock) {
        
        DispatchQueue.main.async {
            var message: [String:Any] = [:]
            var callback: [String:Any] = [:]
            let cardScanSheet = CardScanSheet()
            cardScanSheet.present(from: RCTPresentedViewController()!) { result in
                
                switch result {
                case .completed(var card as ScannedCard?):
                    message["pan"] = card?.pan
                    message["expiryMonth"] =  card?.expiryMonth
                    message["expiryYear"] =  card?.expiryYear
                    callback["status"] = "Succeeded"
                    callback["data"] = message
                case .canceled:
                    callback["status"] = "Cancelled"
                case .failed(let error):
                    callback["status"] = "Failed"
                    
                }
                rnCallback([callback])
            }
        }
    }
}

